samples/matter/manufacturer_specific/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (c) 2025 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})
+
+project(matter-manufacturer_specific)
+
+# Override zap-generated directory.
+get_filename_component(CHIP_APP_ZAP_DIR ${CONFIG_NCS_SAMPLE_MATTER_ZAP_FILES_PATH}/zap-generated REALPATH CACHE)
+
+# Enable GNU STD support and initialize the Matter Data Model.
+include(${ZEPHYR_CONNECTEDHOMEIP_MODULE_DIR}/config/nrfconnect/app/enable-gnu-std.cmake)
```

Comment:
why are we forcing `-std=gnu++17` on app lib through chip module this way ?

This is not portable across toolchains, meaning there is a risk of this breaking when not using gcc.

Zephyr already provides `STD_CPP_VERSION` which is a choice (and thus can be extended), which again uses the compiler property to use the correct flag, meaning each toolchain implementation can either set the appropriate value corresponding to the sample/user request, or error out early if not supported.
Ref: https://github.com/nrfconnect/sdk-zephyr/blob/163004a98b1a59583840215e25f1d876d09d2df5/cmake/compiler/gcc/compiler_flags.cmake#L140
https://github.com/nrfconnect/sdk-zephyr/blob/163004a98b1a59583840215e25f1d876d09d2df5/CMakeLists.txt#L274-L275

Why is the existing infrastructure not being used ?

Reply:
I don't remember the exact reason but some Matter code required `gnu++17` instead of `c++17` in the past and most of the Matter community builds their apps using `gnu++17`, so it seemed the least problematic to use what others do.

I tried to add GNU standards to `STD_CPP` choice but you didn't acknowledge our needs: https://github.com/zephyrproject-rtos/zephyr/pull/42062 so we had to provide a workaround. :). Besides, `-std=gnu++17` is accepted by clang so is the portability a real concern?

Reply:
This part is a copy of the template sample. All Matter samples have been doing this for over four years https://github.com/nrfconnect/sdk-nrf/blob/18e7ef467c2cb2467eaa7b057b25b098298077a4/samples/matter/light_bulb/CMakeLists.txt

We can investigate it as a follow up, but this is out of scope of this PR as we will have to align for all samples.

Reply:
@tejlmand any input on this?

Reply:
@tejlmand CI is green and all other codeowners approved the PR. Your change request refers to a common code for matter samples and is out of scope for this change (as I said yesterday, we can investigate is as follow-up). Please unblock merging the pull request as this now blocks multiple tasks for our team.

Reply:
@tejlmand created jira ticket KRKNWK-20065 to address your comment as a follow-up. Please unblock the PR.